graviti/utility/lazy.py

```diff
@@ -112,6 +113,64 @@ def update(self, pos: int, data: Any) -> None:
             self.pages[pos] = pa.array(self._extractor(data))
 
 
+class MutableLazyList(LazyList[_T]):
+    """MutableLazyList is a lazy load list which follows Sequence protocol.
+
+    It supports extend method to add items into the list.
+
+    Arguments:
+        total_count: The total count of the elements in the lazy list.
+        limit: The size of each lazy load page.
+        fetcher: A callable object to fetch the data and load it to the lazy list.
+        extractor: A callable object to make the source data to an iterable object.
+        dtype: The pyarrow data type of the elements in the lazy list.
+
+    Attributes:
+        pages: A list of pyarrow arrays that contains the data in the lazy list.
+
+    """
+
+    _repr_type = ReprType.SEQUENCE
+
+    def __init__(
+        self,
+        total_count: int,
+        limit: int,
+        fetcher: Callable[[int], None],
+        extractor: Callable[[Any], Iterable[Any]],
+        *,
+        dtype: Optional[pa.DataType] = None,
+    ) -> None:
+        super().__init__(total_count, limit, fetcher, extractor, dtype=dtype)
+        self._offsets = list(range(0, total_count, limit))
+
+    def _getitem(self, index: int) -> _T:
+        """Get item from the lazy list.
+
+        Arguments:
+            index: The index of the item which needs to be non-negative.
+
+        Returns:
+            The item at the index position.
+
+        """
+        page_number = bisect_right(self._offsets, index) - 1
+        offset = self._offsets[page_number]
+        return self.pages[page_number][index - offset]
+
+    def extend(self, values: Iterable[_T]) -> None:
+        """Extend mutable sequence by appending elements from the iterable.
+
+        Arguments:
+            values: Elements to be Extended into the mutable sequence.
+
+        """
+        page = pa.array(values, type=self._dtype)
```

Comment:
We need to build an ExtensionArray when `self._dtype` is the instance of GravitiExtension